mitiq/mitiq_pyquil/benchmarking.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Utilities for generating benchmarking programs.
+"""
+
+from typing import List, Optional, Sequence
+
+import numpy as np
+
+from pyquil import Program
+from pyquil.api import BenchmarkConnection
+from pyquil.gates import CZ, RX, RZ
+from pyquil.quilbase import Gate
+
+NATIVE_1Q_GATES = [RX, RZ]
+MAGIC_ANGLES = [-np.pi, -np.pi / 2, np.pi / 2, np.pi]
+
+
+def one_qubit_gateset(qubit: int) -> List[Gate]:
+    """
+    Return the RX, RZ native gateset on one qubit.
+    """
+    return list(
+        gate(angle, qubit)
+        for gate in NATIVE_1Q_GATES
+        for angle in MAGIC_ANGLES
+    )
+
+
+def two_qubit_gateset(q0: int, q1: int) -> List[Gate]:
+    """
+    Return the RX, RZ, CZ native gateset on two qubits.
+    """
+    return one_qubit_gateset(q0) + one_qubit_gateset(q1) + [CZ(q0, q1)]
+
+
+def generate_rb_program(
```

Comment:
An alternative would be to generate an RB circuit in Qiskit (we have a function for this) and use conversions. This might be preferable as we would be consistent between RB circuits in Qiskit and pyQuil. We could even make it a more general function and specify the output type.

Reply:
This is the function that I used for the paper, so I wanted to include it in the repo. But we can drop if you disagree.

Reply:
I'd say we can keep it for now and open a new issue to consolidate RB functions. What do you think?

If you agree and want to consolidate now, that's fine also, but I don't think it's blocking for this PR. We can include the function elsewhere (mitiq-examples maybe) so the code for the paper is somewhere.

Reply:
Issue opened: #385